tls/s2n_connection.h

```diff
@@ -148,6 +148,9 @@ struct s2n_connection {
      */
     uint16_t max_outgoing_fragment_length;
 
+    /* Negotiated TLS extension Maximum fragment size */
+    uint16_t max_fragment_length;
```

Comment:
I agree that having both `max_outgoing_fragment_length` and `max_fragment_length` can be confusing. 

However, I didn't want to create `max_incoming_fragment_length ` when we currently do not check on incoming fragments. 

The thought process here was to have `max_fragment_length ` as both the flag to show that mfl was set and represent the negotiated value all s2n fragments should follow.